packages/@lwc/engine-core/src/framework/decorators/register.ts

```diff
@@ -266,7 +293,13 @@ export function registerDecorators(
             if (process.env.NODE_ENV !== 'production') {
                 validateObservedField(Ctor, fieldName, descriptor);
             }
-            observedFields[fieldName] = createObservedFieldPropertyDescriptor(fieldName);
+
+            // [W-9927596] Only mark a field as observed when there is no associated public
+            // accessor. This is only here for backward compatibility purposes. This happen whenever
+            // a component defines a class field and a public getter/setter with the same name.
+            if (isUndefined(publicProps) || !(fieldName in publicProps)) {
```

Comment:
This check prevents `publicPropertyDescriptor`(or `Accessor`) descriptors from being overridden by the `observedFieldDescriptor`; therefore, I think of `@track` and `@wire`.

Should this check also handle the case for `@track` fields? Note: I will also say that for `@wire`, though the installed descriptor is the same as for `observedField`, there shouldn't be further consequences.

This test should also be valid:

```js
it('duplicated @track field', () => {
    let Ctor;
    const accessors = [];

    expect(('track fields to react to changes in value') => {
        class DuplicateAccessor extends LightningElement {
            _person;
            @track _person = { name: 'original' };

            @api setPersonName(newName) {
                this._person.name = newName;
            }

            render() {
                return duplicatePropertyTemplate;
            }
        }

        Ctor = DuplicateAccessor;
    }).toLogErrorDev(
        /Invalid observed _person field\. Found a duplicate accessor with the same name\./
    );

    const elm = createElement('x-duplicate-accessor', { is: Ctor });

    document.body.appendChild(elm);

    expect(elm.shadowRoot.querySelector('p').textContent).toBe('original');
    elm.setPersonName('modified');

    return Promise.resolve().then(() => {
        expect(elm.shadowRoot.querySelector('p').textContent).toBe('modified');
    })
});
```

Reply:
Good catch I added tests for duplicated fields as well.